protocols/interfaces.go

```diff
@@ -50,6 +51,11 @@ type ObjectiveEvent struct {
 	BlockNum           uint64
 }
 
+// Storable is an object that can be stored by the store.
+type Storable interface {
+	json.Marshaler
+}
```

Comment:
Probably:

```suggestion
type Storable interface {
	json.Marshaler
	// Id returns a unique identifying key for the Storable item
	Id() string // or bytes or whatever
}
```

Reply:
`Id` is not currently allowed, since `*Channel` needs to be `Storable` and `Channel` has an `Id` field.

If we desire this, I'd suggest it in a follow-up PR.

Reply:
`Key()` is probably better aligned with a `Storable` interface.